zingolib/src/wallet/traits.rs

```diff
@@ -421,14 +406,14 @@ pub trait ReceivedNoteAndMetadata: Sized {
     fn from_parts(
         diversifier: Self::Diversifier,
         note: Self::Note,
-        witness_position: Position,
+        witness_position: Option<Position>,
```

Comment:
I don't think this field name makes sense..    the position of a value-commitment is part of what is witnessed.

Reply:
Isn't this an input to a witness construction procedure?

Reply:
I suspect this name was a typo...and tat it was always supposed to be 'witnessed_position'...but a better name would make clear that it's the position of the commitment that we need to be able to create a witness to.